dask/dataframe/csv.py

```diff
@@ -40,7 +40,15 @@ def bytes_read_csv(b, header, kwargs, dtypes=None, columns=None,
         bio.write(header)
     bio.write(b)
     bio.seek(0)
-    df = pd.read_csv(bio, **kwargs)
+
+    try:
+        df = pd.read_csv(bio, **kwargs)
+    except (csv.Error, pd.io.common.CParserError) as e:
+        raise ValueError(
+            "There was an error in parsing a CSV block. This may be due to "
+            "the line terminator appearing in a quoted value. Original error: "
+            "%s" % str(e))
```

Comment:
Yeah, there certainly could be other reasons why a CSV file might not parse.

The exception for the C engine is `Error tokenizing data. C error: EOF inside string starting at line 1`.  For the  python engine it's  `',' expected after '"'`, which is pretty vague.

I could just have it only add the note about line terminators if `"EOF inside string"` is in the original exception text, and otherwise reraise the original exception.  That wouldn't help when using the python engine, but that's probably a relatively rare use case.

Thoughts?